### PR TITLE
TASK: Localize Date Picker (refs #1534)

### PR DESCRIPTION
**Pull request by lorenzulrich**

### CODE REVIEW

**Comment by dfeyer** [comment]
@lorenzulrich Happy to see your first contribution here ;)

Did you check if those labels exists for the old UI, if yes, just use the existing IDs

**Comment by lorenzulrich** [comment]
@dfeyer Thanks ;-).

The label "today" doesn't exist, the label "Apply" does, but in a different context (applying changes made in the inspector). Are the labels used generally or context-specific?

**Comment by mstruebing** [comment]
I would suggest to introduce a new label for the Datepicker-Apply to not mix contexts with the apply button. Make changes in the future easier.

**Comment by mstruebing** [comment]
@lorenzulrich Please rebase your branch against the latest master, the master is repaired so that travis will run through if everything is fine

**Comment by lorenzulrich** [comment]
Comment from @mstruebing:

> It's something with the dependencies fucked up. `lerna ERR! execute     Cannot find module '@neos-project/neos-ui-decorators' from 'dateInput.js'`

> Maybe it's enoug to add this dependencie to this subpackage, but I need to try.

**Comment by lorenzulrich** [comment]
@mstruebing I think I did everything I could but it still doesn't work out. To me, it seems that the problem is that `react-ui-components` seems to be a kind of "standalone" package with an own Git repository:

```
  "repository": {
    "type": "git",
    "url": "https://github.com/neos/react-ui-components.git"
  },
```

(Which throws a 404 - is it maybe a protected repo?)

So this package is not aware of the packages that are not on NPM, but just a part of `Neos.Neos.Ui`.

**Comment by mstruebing** [comment]
@lorenzulrich: I looked at your PR and it seems lerna isn't doing what it should. The `@neos-project/neos-ui-i18n` dependency should be in the `node_modules` of the
`react-ui-components` package, but isn't. So when I symlinked the package manually inside the `node_modules` the tests are running through. However, I've asked @grebaldi how to
handle this with lerna and the first thing was that the `react-ui-components` should not know about this, they are only presentational and didn't know anything, they get
 everything they need from outside. So the correct way should be to add the needed labels to the Props and give them in from the outside, in this case:
`neos-ui-editors/src/editors/DateTime` (that is already connected to the store for example).

This is also the case for https://github.com/neos/neos-ui/pull/1537

File: packages/react-ui-components/src/DateInput/dateInput.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import PropTypes from 'prop-types';
 import enhanceWithClickOutside from 'react-click-outside';
 import moment from 'moment';
 import mergeClassNames from 'classnames';
+import I18n from '@neos-project/neos-ui-i18n';
```
Comment (mstruebing) [inline packages/react-ui-components/src/DateInput/dateInput.js]: Please remove this dependency and pass the needed labels from the outside (`neos-ui-editors`)

File: packages/react-ui-components/src/DateInput/dateInput.js [hunk 2]
```diff
@@ -96,8 +95,6 @@ export class DateInput extends PureComponent {
             theme,
             value,
             id,
-            applyLabel,
-            todayLabel,
```
Comment (mstruebing) [inline packages/react-ui-components/src/DateInput/dateInput.js]: This should be readded, sorry!

**Comment by lorenzulrich** [comment]
Yeah! Thanks to your explanations, I now understand that the `react-ui-components` are presentational only. It seems that this change is now looking good and I hope it will get merged.

**Comment by kitsunet** [comment]
@lorenzulrich The respective labels used here do not exist yet, right? Is there a corresponding PR in the neos-dev-collection?

**Comment by lorenzulrich** [comment]
@kitsunet see here: https://github.com/neos/neos-development-collection/pull/1851

**Comment by kitsunet** [comment]
Now this conflicts after the other change is merged. But if you can fix it quickly @lorenzulrich I would merge right after :)

**Comment by kitsunet** [comment]
Waiting for travis here.

**Pull request merged**